tests/sitl/__main__.py

```diff
@@ -0,0 +1,138 @@
+#!/usr/bin/env python -u
+
+from __future__ import print_function
+import time
+import os
+from subprocess import Popen, PIPE
+import atexit
+import sys
+import tempfile
+from functools import wraps
+import errno
+import os
+import signal
+import time
+import psutil
+
+def wait_timeout(proc, seconds):
+    """Wait for a process to finish, or raise exception after timeout"""
+    start = time.time()
+    end = start + seconds
+    interval = min(seconds / 1000.0, .25)
+
+    while True:
+        result = proc.poll()
+        if result is not None:
+            return result
+        if time.time() >= end:
+            raise RuntimeError("Process timed out")
+        time.sleep(interval)
+
+def kill(proc_pid):
+    process = psutil.Process(proc_pid)
+    for proc in process.children(recursive=True):
+        try:
+            proc.kill()
+        except psutil.NoSuchProcess:
+            pass
+    try:
+        process.kill()
+    except psutil.NoSuchProcess:
+        pass
+
+if sys.platform == "win32":
+    import msvcrt
+    import _subprocess
+else:
+    import fcntl
+
+bg = []
+def cleanup_processes():
+    for p in bg:
+        kill(p.pid)
+atexit.register(cleanup_processes)
+
+testpath = os.path.dirname(__file__)
+
+
+def wrap_fd(pipeout):
+    # Prepare to pass to child process
+    if sys.platform == "win32":
+        curproc = _subprocess.GetCurrentProcess()
+        pipeouth = msvcrt.get_osfhandle(pipeout)
+        pipeoutih = _subprocess.DuplicateHandle(curproc, pipeouth, curproc, 0, 1, _subprocess.DUPLICATE_SAME_ACCESS)
+        return (str(int(pipeoutih)), pipeoutih)
+    else:
+        return (str(pipeout), None)
+
+def lets_run_a_test(name):
+    sitl = Popen(['dronekit-sitl', 'copter-3.3-rc5', '-I0', '-S', '--model', 'quad', '--home=-35.363261,149.165230,584,353'], stdout=PIPE, stderr=PIPE)
```

Comment:
maybe handle this on a config?, not clear to me why we are testing to a release candidate instead of a stable version, handling this on a config would help bumping the version once we are tied to an Ardupilot version, also people are probably going to ask about testing for Plane and Rover